site/ar/tutorials/text/nmt_with_attention.ipynb

```diff
@@ -0,0 +1,1136 @@
+    "  # creating cleaned input, output pairs\n",
+    "  targ_lang, inp_lang = create_dataset(path, num_examples)\n",
+    "\n",
+    "  input_tensor, inp_lang_tokenizer = tokenize(inp_lang)\n",
+    "  target_tensor, targ_lang_tokenizer = tokenize(targ_lang)\n",
+    "\n",
+    "  return input_tensor, target_tensor, inp_lang_tokenizer, targ_lang_tokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "GOi42V79Ydlr"
+   },
+   "source": [
+    "### <div dir=\"rtl\"> تحديد حجم مجموعة البيانات للتجربة بشكل أسرع (اختياري)</div>\n",
+    "\n",
+    "<div dir=\"rtl\"> \n",
+    "\n",
+    "سيستغرق التدريب على مجموعة البيانات الكاملة (أكثر من 100،000 جملة) وقتًا طويلاً. للتدريب بشكل أسرع ، يمكننا تحديد حجم مجموعة البيانات إلى 30،000 جملة (بالطبع ، تنقص جودة الترجمة ببيانات أقل):\n",
+    "\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "cnxC7q-j3jFD"
+   },
+   "outputs": [],
+   "source": [
+    "# Try experimenting with the size of that dataset\n",
+    "num_examples = 30000\n",
+    "input_tensor, target_tensor, inp_lang, targ_lang = load_dataset(path_to_file, num_examples)\n",
+    "\n",
+    "# Calculate max_length of the target tensors\n",
+    "max_length_targ, max_length_inp = max_length(target_tensor), max_length(input_tensor)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "4QILQkOs3jFG"
+   },
+   "outputs": [],
+   "source": [
+    "# Creating training and validation sets using an 80-20 split\n",
+    "input_tensor_train, input_tensor_val, target_tensor_train, target_tensor_val = train_test_split(input_tensor, target_tensor, test_size=0.2)\n",
+    "\n",
+    "# Show length\n",
+    "print(len(input_tensor_train), len(target_tensor_train), len(input_tensor_val), len(target_tensor_val))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "lJPmLZGMeD5q"
+   },
+   "outputs": [],
+   "source": [
+    "def convert(lang, tensor):\n",
+    "  for t in tensor:\n",
+    "    if t!=0:\n",
+    "      print (\"%d ----> %s\" % (t, lang.index_word[t]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "VXukARTDd7MT"
+   },
+   "outputs": [],
+   "source": [
+    "print (\"Input Language; index to word mapping\")\n",
+    "convert(inp_lang, input_tensor_train[0])\n",
+    "print ()\n",
+    "print (\"Target Language; index to word mapping\")\n",
+    "convert(targ_lang, target_tensor_train[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "rgCLkfv5uO3d"
+   },
+   "source": [
+    "### <div dir=\"rtl\"> أنشئ مجموعة بيانات من نوع tf.data </div> "
```

Comment:
As I suggested in this PR, we need to clarify the english subtitle https://github.com/tensorflow/docs/pull/1515. it will eventually be something like this

```
Use tf.data to create batches and shuffle the dataset
```

so the arabic translation would be 

<div dir="rtl">

استعمل tf.data لخلط البيانات و توزيعها على دفعات

</div>

Reply:
That is an open PR. Will include  this when it's approved.